deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -839,8 +844,12 @@ def post_add_stage(request, name):
         else:
             common.create_simple_stage(request,name, stage, description, external_id)
     except:
-        environs_helper.delete_nimbus_identifier(request, external_id)
-        raise
+        try:
+            environs_helper.delete_nimbus_identifier(request, external_id)
+        except TeletraanException as detail:
+            message = 'Failed to delete Nimbus identifier {}, Error Message: {}'.format(external_id, detail)
+            log.error(message)
+            messages.add_message(request, messages.ERROR, message)
```

Comment:
This implementation doesn't propagate the original issue to the user. For example, if `clone_from_stage_name` fails before it created a nimbus identifier and results in an exception `originalException`, this call `delete_nimbus_identifier` is expected to fail as well. In this case `originalException` is more important than `detail`.

Reply:
I see what you mean. I'll update it.